server/cmd/app.go

```diff
@@ -96,6 +97,49 @@ func (app *Application) listenAndServe() {
 	}
 }
 
+func (app *Application) handleRecommendationsTrends(w http.ResponseWriter, r *http.Request) {
+	params := r.URL.Query()
+
+	start, err := time.Parse(time.RFC3339, params.Get("startDateTime"))
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		doResponse(w, false, "startDateTime must be a valid ISO8601 string")
+		return
+	}
+
+	end, err := time.Parse(time.RFC3339, params.Get("endDateTime"))
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		doResponse(w, false, "endDateTime must be a valid ISO8601 string")
+		return
+	}
+
+	if end.Before(start) {
+		w.WriteHeader(http.StatusBadRequest)
+		doResponse(w, false, "endDateTime must be after startDateTime")
+		return
+	}
+
```

Comment:
We should set a limit on the time range of query (e.g. `end - start`).

Reply:
limited the diff/range between 1 sec to 1 day